galsim/image.py

```diff
@@ -475,21 +574,232 @@ def setSubImage(self, bounds, rhs):
+        the most negative value.  For example,
+
+            >>> N = 100
+            >>> im_full = galsim.ImageC(bounds=galsim.BoundsI(0,N/2,-N/2,N/2), scale=dk)
+            >>> # ... fill with im[i,j] = FT(kx=i*dk, ky=j*dk)
+            >>> N2 = 64
+            >>> im_wrap = im_full.wrap(galsim.BoundsI(0,N/2,-N2/2,N2/2-1, hermitian='x')
+
+        This sets up im_wrap to be the properly Hermitian version of the data appropriate for
+        passing to an FFT.
+
+        Note that this routine modifies the original image (and not just the subimage onto which
+        it is wrapped), so if you want to keep the original pristine, you should call
+        `wrapped_image = image.copy().wrap(bounds)`.
+
+        @param bounds       The bounds of the subimage onto which to wrap the full image.
+        @param hermitian    Whether the image is implicitly Hermitian and if so, whether it is the
+                            x or y values that are not stored.  [default: False]
+
+        @returns the subimage, image[bounds], after doing the wrapping.
+        """
+        if not isinstance(bounds, galsim.BoundsI):
+            raise TypeError("bounds must be a galsim.BoundsI instance")
+        if not hermitian:
+            subimage = self.image.wrap(bounds, False, False)
+        elif hermitian == 'x':
+            if self.bounds.xmin != 0:
+                raise ValueError("hermitian == 'x' requires self.bounds.xmin == 0")
+            if bounds.xmin != 0:
+                raise ValueError("hermitian == 'x' requires bounds.xmin == 0")
+            subimage = self.image.wrap(bounds, True, False)
+        elif hermitian == 'y':
+            if self.bounds.ymin != 0:
+                raise ValueError("hermitian == 'y' requires self.bounds.ymin == 0")
+            if bounds.ymin != 0:
+                raise ValueError("hermitian == 'y' requires bounds.ymin == 0")
+            subimage = self.image.wrap(bounds, False, True)
+        else:
+            raise ValueError("Invalid value for hermitian: %s"%hermitian)
+        return Image(image=subimage, wcs=self.wcs)
+
+    def calculate_fft(self):
+        """Performs an FFT of an Image in real space to produce a k-space Image.
+
+        Note: the image will be padded with zeros as needed to make an image with bounds that
+        look like BoundsI(-N/2, N/2-1, -N/2, N/2-1).
+
+        The input image must have a PixelScale wcs.  The output image will be complex (an ImageC
+        instance) and its scale will be 2pi / (N dx), where dx is the scale of the input image.
+
+        @returns an ImageC instance with the k-space image.
         """
-        self.setSubImage(bounds,rhs)
+        if self.wcs is None:
+            raise ValueError("calculate_fft requires that the scale be set.")
+        if not self.wcs.isPixelScale():
+            raise ValueError("calculate_fft requires that the image has a PixelScale wcs.")
```

Comment:
No, because later we'll call `dk = self.scale`, which would raise an exception if it's not either PixelScale or OffsetWCS.  We'd need special code to handle a JacobianWCS with a jacobian that happens to be a multiple of the identity matrix.  (Plus, `wcs.compaible` wouldn't work to determine that, since what would you compare it to?)